src/Game.js

```diff
@@ -74,25 +74,42 @@ const playCard = (G, ctx, card) => {
     console.log(`${ctx.playerID} played`);
     const playerId = ctx.playerID;
     const playedCard = G.players[playerId].cards[card];
-    G.players[playerId].cards = G.players[playerId].cards.filter(c => c !== playedCard);
-    G.aboutToPlay[playerId] = null;
-
-    G.cardsPlayed.push(playedCard);
-    G.gameOver = isGameOver(G);
+    return {
+        ...G,
+        players: {
+            [playerId]: {
+                cards: G.players[playerId].cards.filter(c => c !== playedCard)
+            }
+        },
+        aboutToPlay: {
+            [playerId]: null
+        },
+        cardsPlayed: [...G.cardsPlayed, playedCard],
+        gameOver: isGameOver(G)
+    }
 };
 
 const aboutToPlay = (G, ctx) => {
-    G.aboutToPlay[ctx.playerID] = Date.now();
+    return {
+        ...G,
+        aboutToPlay: {
+            [ctx.playerID]: Date.now()
+        }
+    }
 };
 
 const cancelPlay = (G, ctx) => {
-    G.aboutToPlay[ctx.playerID] = null;
+    return {
+        ...G,
+        aboutToPlay: {
+            [ctx.playerID]: null
+        }
```

Comment:
same as above, this deletes the entire contents `aboutToPlay`...can you change it to something like this?
```suggestion
        aboutToPlay: {
            ...aboutToPlay
            [playerId]: null
        },
```

Reply:
Gotcha! Fixed.